src/goto-symex/goto_symex_state.cpp

```diff
@@ -618,6 +644,20 @@ void goto_symex_statet::rename_address(
   }
 }
 
+/// Return the type corresponding to a symbol or tag. Empty optional if \p type
+/// is neither a symbol or a tag.
+static optionalt<typet>
+type_of_symbol_or_tag(const typet &type, const namespacet &ns)
+{
+  if(type.id() == ID_symbol_type)
+    return ns.lookup(to_symbol_type(type)).type;
+  if(type.id() == ID_union_tag)
+    return ns.lookup(to_union_tag_type(type)).type;
+  if(type.id() == ID_struct_tag)
+    return ns.lookup(to_struct_tag_type(type)).type;
+  return {};
+}
+
```

Comment:
This should not be necessary.

Reply:
If you mean "renaming is never needed for tagged types," I think that's untrue when we have structs with a flexible array member to rename.